### ISSUE TITLE
Subquery splicing leaks variable into COLLECT … INTO result

### ISSUE THREAD

**Issue by Simran-B** [arangodb/arangodb#10896]
## My Environment

* __ArangoDB Version__:        devel @ 847175ebc7ac4153b89b2d7a08f71cf69ae54d1d
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Single Server
* __Deployment Strategy__:   Manual Start
* __Configuration__:               default
* __Infrastructure__:               own
* __Operating System__:        Ubuntu 18.04
* __Used Package__:              self-compiled

## Component, Query & Data

__Affected feature__:
AQL

__AQL query (if applicable)__:

```js
LET tokens = (
  FOR word IN TOKENS("Some Text", "text_en")
    FILTER word IN TOKENS("Some Other Thing", "text_en")
    RETURN word
)
COLLECT token = tokens INTO bucket
//LIMIT 1
RETURN bucket[0]
```

__AQL explain (if applicable)__:

```js
Execution plan:
 Id   NodeType            Est.   Comment
  1   SingletonNode          1   * ROOT
 12   SubqueryStartNode      1     - LET tokens = ( /* subquery begin */
  3   CalculationNode        1       - LET #6 = [ "some", "text" ]   /* json expression */   /* const assignment */
  4   EnumerateListNode      2       - FOR word IN #6   /* list iteration */
  5   CalculationNode        2         - LET #8 = (word in [ "some", "other", "thing" ])   /* simple expression */
  6   FilterNode             2         - FILTER #8
 13   SubqueryEndNode        2     - RETURN  word ) /* subquery end */
 11   SortNode               2     - SORT tokens ASC   /* sorting strategy: standard */
  9   CollectNode            2     - COLLECT token = tokens INTO bucket   /* sorted */
 10   ReturnNode             2     - RETURN bucket

Indexes used:
 none

Optimization rules applied:
 Id   RuleName
  1   splice-subqueries
```

## Steps to reproduce

1. Run above query
2. See the `word` attribute and that splice-subqueries optimization rule is applied
3. Uncomment `// LIMIT 1` as a quick way to disable the rule, re-run and see that `word` is no longer present

__Problem__:

You will see **(non-representable type none)** in the Web UI and **undefined** in arangosh as `word`'s value for the original query, whereas it is missing if splicing is not applied.

```json
[
  {
    "word": "(non-representable type none)",
    "tokens": [
      "some"
    ]
  }
]
```

__Expected result__:

The `word` attribute shouldn't leak outside the subquery I guess, even with the splice-subqueries optimization on.

FWIW `… RETURN bucket[0].word` returns `[ null ]`.

**Comment by goedderz** [comment]
Fixed in https://github.com/arangodb/arangodb/pull/10945 / https://github.com/arangodb/arangodb/pull/10959.

**Comment by Simran-B** [comment]
Tried in devel, looking good. Thanks!

**Comment by maxkernbach** [comment]
Closing - Fixed in 3.6.1.
